utility/utils.py

```diff
@@ -1884,12 +1884,34 @@ def clone_the_repo(config, node, path_to_clone):
 
 
 def calculate_available_storage(node):
-    log.info("Calculate current storage present in cluster")
-    out, err = node.exec_command(cmd="sudo ceph df --format json")
     import json
 
+    log.info(
+        "Calculate maximum storage that is available to be used."
+        + " It is the amount of data that can be used before the first OSD becomes full."
+        + " This is implicitly divided by replication factor or erasure code"
+    )
+    out, err = node.exec_command(cmd="sudo radosgw-admin zone get --format json")
     out = json.loads(out)
-    return out["stats"]["total_avail_bytes"]
+    zone_name = out["name"]
+    rgw_bucket_data_pool = f"{zone_name}.rgw.buckets.data"
+    out, err = node.exec_command(cmd="sudo ceph df --format json")
+    if rgw_bucket_data_pool not in out:
+        node.exec_command(cmd=f"sudo ceph osd pool create {rgw_bucket_data_pool} 64 64")
+        time.sleep(10)
+        out, err = node.exec_command(cmd="sudo ceph df --format json")
+    ceph_df_json = json.loads(out)
+    # Ceph uses below given formula to calculate MAX AVAIL value :
+    # [min(osd.avail for osd in OSD_up) - ( min(osd.avail for osd in OSD_up).total_size * (1 - mon_osd_full_ratio)) ]
+    #   * len(osd.avail for osd in OSD_up) /pool.size()
+    # min(osd.avail for osd in OSD_up) : Minimum space left in an OSD in up set in pool crush ruleset.
+    #   your usage is bounded by osd.X.
+    # len(osd.avail for osd in OSD_up) : Number of OSDs in UP set in pool crush ruleset
+    # pool.size() : pool replication size
+    # refer https://access.redhat.com/solutions/2273951
+    for pool in ceph_df_json["pools"]:
+        if pool["name"] == rgw_bucket_data_pool:
+            return pool["stats"]["max_avail"]
```

Comment:
Since code already creates expected pool, please return `max_avail` directly without looping it

Reply:
@udaysk23 , as discussed the ceph_df_json["pools"] is a list of dictionaries, so looping through it to get to the right pool based on pool_name conditional. below is sample ceph df json output
```
{
    "stats": {
        "total_bytes": 343530274816,
        "total_avail_bytes": 312613064704,
        "total_used_bytes": 30917210112,
        "total_used_raw_bytes": 30917210112,
        "total_used_raw_ratio": 0.089998498558998108,
        "num_osds": 16,
        "num_per_pool_osds": 16,
        "num_per_pool_omap_osds": 16
    },
    "stats_by_class": {
        "hdd": {
            "total_bytes": 343530274816,
            "total_avail_bytes": 312613064704,
            "total_used_bytes": 30917210112,
            "total_used_raw_bytes": 30917210112,
            "total_used_raw_ratio": 0.089998498558998108
        }
    },
    "pools": [
        {
            "name": "primary.rgw.buckets.index",
            "id": 10,
            "stats": {
                "stored": 4323901,
                "objects": 407,
                "kb_used": 12668,
                "bytes_used": 12971704,
                "percent_used": 4.6608762204414234e-05,
                "max_avail": 92765814784
            }
        },
        {
            "name": "primary.rgw.buckets.data",
            "id": 11,
            "stats": {
                "stored": 8909014016,
                "objects": 55957,
                "kb_used": 26427336,
                "bytes_used": 27061592064,
                "percent_used": 0.088622212409973145,
                "max_avail": 92765814784
            }
        },
        {
            "name": "primary.rgw.buckets.non-ec",
            "id": 12,
            "stats": {
                "stored": 3475,
                "objects": 0,
                "kb_used": 11,
                "bytes_used": 10426,
                "percent_used": 3.7463514956925792e-08,
                "max_avail": 92765814784
            }
        }
    ]
}
```